tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    # Bright change gets accepted.
+    brightness = 10
+    client.adjustment = [{const.KEY_BRIGHTNESS: brightness}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["brightness"] == round(255 * (brightness / 100.0))
+
+    # Broken brightness value is ignored.
+    bad_brightness = -200
+    client.adjustment = [{const.KEY_BRIGHTNESS: bad_brightness}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["brightness"] == round(255 * (brightness / 100.0))
+
+    # Update components.
+    client.is_on.return_value = True
+    call_registered_callback(client, "components-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "on"
+
+    client.is_on.return_value = False
+    call_registered_callback(client, "components-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "off"
+
+    # Update priorities (V4L)
+    client.is_on.return_value = True
+    client.visible_priority = {const.KEY_COMPONENTID: const.KEY_COMPONENTID_V4L}
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_EXTERNAL_SOURCE
+    assert entity_state.attributes["hs_color"] == (0.0, 0.0)
+    assert entity_state.attributes["effect"] == const.KEY_COMPONENTID_V4L
+
+    # Update priorities (Effect)
+    effect = "foo"
+    client.visible_priority = {
+        const.KEY_COMPONENTID: const.KEY_COMPONENTID_EFFECT,
+        const.KEY_OWNER: effect,
+    }
+
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["effect"] == effect
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_EFFECT
+    assert entity_state.attributes["hs_color"] == (0.0, 0.0)
+
+    # Update priorities (Color)
+    rgb = (0, 100, 100)
+    client.visible_priority = {
+        const.KEY_COMPONENTID: const.KEY_COMPONENTID_COLOR,
+        const.KEY_VALUE: {const.KEY_RGB: rgb},
+    }
+
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["effect"] == hyperion_light.KEY_EFFECT_SOLID
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_LIGHTBULB
+    assert entity_state.attributes["hs_color"] == (180.0, 100.0)
+
+    # Update effect list
+    effects = [{const.KEY_NAME: "One"}, {const.KEY_NAME: "Two"}]
+    client.effects = effects
+    call_registered_callback(client, "effects-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["effect_list"] == [
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6233107)